server/src/metadata/commands/sync-tenant-metadata.command.ts

```diff
@@ -0,0 +1,63 @@
+import { Command, CommandRunner, Option } from 'nest-commander';
+
+import { ObjectMetadataService } from 'src/metadata/object-metadata/services/object-metadata.service';
+import { FieldMetadataService } from 'src/metadata/field-metadata/services/field-metadata.service';
+import { TenantInitialisationService } from 'src/metadata/tenant-initialisation/tenant-initialisation.service';
+import { DataSourceMetadataService } from 'src/metadata/data-source-metadata/data-source-metadata.service';
+
+// TODO: implement dry-run
+interface RunTenantMigrationsOptions {
+  workspaceId: string;
+}
+
+@Command({
+  name: 'tenant:sync-metadata',
+  description: 'Sync metadata',
+})
+export class SyncMetadata extends CommandRunner {
+  constructor(
+    private readonly objectMetadataService: ObjectMetadataService,
+    private readonly fieldMetadataService: FieldMetadataService,
+    private readonly dataSourceMetadataService: DataSourceMetadataService,
+    private readonly tenantInitialisationService: TenantInitialisationService,
+  ) {
+    super();
+  }
+
+  async run(
+    _passedParam: string[],
+    options: RunTenantMigrationsOptions,
+  ): Promise<void> {
+    // TODO: run in a dedicated job + run queries in a transaction.
+    const dataSourceMetadata =
+      await this.dataSourceMetadataService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        options.workspaceId,
+      );
+
+    // TODO: This solution could be improved, using a diff for example, we should not have to delete all metadata and recreate them.
+    await this.fieldMetadataService.deleteMany(
+      { workspaceId: { eq: options.workspaceId } },
+      { useSoftDelete: false },
```

Comment:
This useSoftDelete does not work as I expected. It's still softDeleting our objects and we don't want that here... I had to temporarily deactivate this default behaviour inside fieldMetadataService and objectMetadataService

Reply:
@Weiko As you disabled `softDelete` is it still needed ?

Reply:
It's not @magrinj good point, I'll remove that 🙂